lasagne/regularization.py

```diff
@@ -1,15 +1,47 @@
 import theano.tensor as T
+from .layers import Layer, get_all_params
 
-from . import layers
 
+def l1(param):
+    """ L1 weight decay penalty """
+    return T.sum(abs(param))
 
-def l2(layer, include_biases=False):
-    if include_biases:
-        all_params = layers.get_all_params(layer)
-    else:
-        all_params = layers.get_all_non_bias_params(layer)
 
-    return sum(T.sum(p**2) for p in all_params)
+def l2(param):
+    """ L2 weight decay penalty """
+    return T.sum(param**2)
 
 
-# TODO: sparsity regularization
+def apply_penalty(params, penalty, **kwargs):
+    try:
+        if len(params) == 0:
+            return 0
+        else:
+            return sum(penalty(p, **kwargs) for p in params)
```

Comment:
Can you remove the `len` check? `sum` returns zero for an empty sequence anyway, and using `len` will require `params` to have a length -- it's more flexible if we allow generators as well.